src/chain/ethereum.rs

```diff
@@ -189,28 +196,43 @@ impl PevmChain for PevmEthereum {
     // https://github.com/ethereum/go-ethereum/blob/master/cmd/era/main.go#L289
     fn calculate_receipt_root(
         &self,
-        _spec_id: SpecId,
+        spec_id: SpecId,
         txs: &BlockTransactions<Self::Transaction>,
         tx_results: &[PevmTxExecutionResult],
-    ) -> B256 {
-        // 1. Create an iterator of ReceiptEnvelope
-        let tx_type_iter = txs
-            .txns()
-            .map(|tx| TxType::try_from(tx.transaction_type.unwrap_or_default()).unwrap());
+    ) -> Result<Option<B256>, Self::CalculateReceiptRootError> {
+        if spec_id < SpecId::BYZANTIUM {
+            // We can only calculate the receipts root from Byzantium.
+            // Before EIP-658 (https://eips.ethereum.org/EIPS/eip-658), the
+            // receipt root is calculated with the post transaction state root,
+            // which we don't have in these tests.
+            return Ok(None);
```

Comment:
This should be an error, a proper `Ok` would require an extra data type here. Can also promote the comment to a `TODO` (now that we're actively looking for community contributions).

Reply:
can you check my latest commit? here is the modified part:

```rust
        if spec_id < SpecId::BYZANTIUM {
            // We can only calculate the receipts root from Byzantium.
            // Before EIP-658 (https://eips.ethereum.org/EIPS/eip-658), the
            // receipt root is calculated with the post transaction state root,
            // which we don't have in these tests.

            // TODO: allow to calculate the receipt root by providing the post transaction state root
            return Err(CalculateReceiptRootError::Unsupported);
        }
```